library/manageiq_custom_attributes.py

```diff
@@ -168,7 +177,8 @@ def add_or_update_custom_attributes(self, entity_type, entity_name, custom_attri
 
         entity_cas = self.get_entity_custom_attributes(entity_type, entity_id)
         for new_ca in custom_attributes:
-            existing_ca = next((ca for ca in entity_cas if ca['name'] == new_ca['name']), None)
+            existing_ca = next((ca for ca in entity_cas if
+                ManageIQCustomAttributes.compare_custom_attributes(ca, new_ca)), None)
```

Comment:
No, just shorter.  Matters if a subclass would override it, which is irrelevant here.
(`self.` does "virtual" lookup, class name is hard-coded.  
C++ and Java, for unclear reasons, conflated the concept of class methods — that don't receive an instance — with compile-time "static" method lookup.  Python `@staticmethod` while called "static" is actually like `@classmethod` in doing virtual lookup when called from an instance.)